packages/edit-site/src/components/actions/index.js

```diff
@@ -411,3 +412,77 @@ export const postRevisionsAction = {
 		document.location.href = href;
 	},
 };
+
+export const renamePostAction = {
+	id: 'rename-post',
+	label: __( 'Rename' ),
+	RenderModal: ( { items, closeModal } ) => {
```

Comment:
Does this need an `isEligible` check?

Reply:
Yes, it will need to check thrash status:
```
isEligible( post ) {
	return post.status !== 'trash';
},
```

Reply:
Thanks for the confirmation 👍 

It looks like we have a small issue on the site editor's page details screen as it still shows a Move to Trash option for an already trashed page.

Reply:
Good catch! The plan is to consolidate these actions(details, sidebar and data views) and it seems when the original delete action in details was implemented, there was no way to view pages in with trash status. So actually I think we shouldn't allow to edit `trashed` pages, which is the bigger issue.